app/src/main/java/com/github/epfl/meili/poi/PoiGoogleRetriever.kt

```diff
@@ -0,0 +1,112 @@
+package com.github.epfl.meili.poi
+
+import android.util.Log
+import com.android.volley.Request
+import com.android.volley.RequestQueue
+import com.android.volley.VolleyError
+import com.android.volley.toolbox.JsonObjectRequest
+import com.github.epfl.meili.MainApplication
+import com.github.epfl.meili.R
+import com.github.epfl.meili.models.PointOfInterest
+import com.github.epfl.meili.util.HttpRequestQueue
+import com.google.android.gms.maps.model.LatLng
+import com.google.gson.Gson
+import com.google.gson.annotations.SerializedName
+import org.json.JSONObject
+
+open class PoiGoogleRetriever {
+    private var queue: RequestQueue = HttpRequestQueue.getQueue()
+
+    fun setQueue(newQueue: RequestQueue) {
+        queue = newQueue
+    }
+
+    open fun requestPoisAPI(latLng: LatLng?, onSuccess: ((List<PointOfInterest>) -> Unit)?, onError: ((VolleyError) -> Unit)?) {
+
+        if (latLng != null && onSuccess != null && onError != null) {
+            val apiKey = MainApplication.applicationContext().getString(R.string.google_api_key)
+            val query = "location=${latLng.latitude},${latLng.longitude}&radius=$MAX_COVERAGE_RADIUS&type=point_of_interest&key=$apiKey"
+            queryGooglePlacesAPI(query, onSuccess, onError)
+        }
+    }
+
+    fun queryGooglePlacesAPI(query: String, onSuccess: (List<PointOfInterest>) -> Unit, onError: (VolleyError) -> Unit) {
+
+        val jsonObjectRequest = JsonObjectRequest(Request.Method.GET, GOOGLE_PLACES_URL + query,
+                null, customOnSuccessFrom(onSuccess), onError)
+
+        queue.add(jsonObjectRequest)
+    }
+
+    fun customOnSuccessFrom(onSuccess: (List<PointOfInterest>) -> Unit): (JSONObject) -> Unit {
+        return { response ->
+            if (response["status"] == "OK") {
+                val placesResponse = Gson().fromJson(response.toString(), GooglePlacesResponse::class.java)
+                Log.d(TAG, response.toString())
+                Log.d(TAG, placesResponse.getCustomPois().toString())
+                onSuccess(placesResponse.getCustomPois())
+            }
+        }
+    }
+
+    companion object {
+        private const val GOOGLE_PLACES_URL = "https://maps.googleapis.com/maps/api/place/nearbysearch/json?"
+        private const val MAX_COVERAGE_RADIUS = 3000
+        private const val TAG = "PoiGoogleRetriever"
+    }
+}
+
+data class GooglePlacesResponse(
+        @SerializedName("results")
+        val pointsOfInterest: List<PlacesPointOfInterest> = ArrayList()
+) {
```

Comment:
Is there a strong reason for the existence of this class? A class with a single attribute looks unnecessary to me.

Reply:
Again, I am only using these classes for the json object deserialisation, so this depends on how the object has been formed. Moreover, it is possible that for some of these classes (not GooglePlacesResponse but PoiOpeningHours) that the Json provides more attributes. However, they didn't seem important for the purpose of our application and hence I am not retrieving them.

Reply:
I am going to add a comment specifying that these data classes are only for deserialisation (after fetching the info the answer is transformed into a List<PointOfInterest>).